lib/semian/circuit_breaker.rb

```diff
@@ -1,13 +1,35 @@
 module Semian
-  class CircuitBreaker
-    attr_reader :state
-
-    def initialize(exceptions:, success_threshold:, error_threshold:, error_timeout:)
+  class CircuitBreaker #:nodoc:
+    def initialize(name, exceptions:, success_threshold:, error_threshold:, error_timeout:, permissions: 0660)
+      @name = name.to_s
       @success_count_threshold = success_threshold
       @error_count_threshold = error_threshold
       @error_timeout = error_timeout
       @exceptions = exceptions
-      reset
+
+      @sliding_window = ::Semian::SysVSlidingWindow.new(
+        "#{name}_sliding_window",
+        @error_count_threshold,
+        permissions)
+      @successes = ::Semian::SysVAtomicInteger.new(
+        "#{name}_atomic_integer",
+        permissions)
+      @state = ::Semian::SysVAtomicEnum.new(
```

Comment:
I think you need to split this PR in two yet again, and let the SysV stuff be reviewed independently.